src/vs/workbench/services/userDataSync/browser/userDataSyncWorkbenchService.ts

```diff
@@ -399,32 +399,41 @@ export class UserDataSyncWorkbenchService extends Disposable implements IUserDat
 	}
 
 	private async handleConflictsWhileTurningOn(token: CancellationToken): Promise<void> {
-		const result = await this.dialogService.show(
-			Severity.Warning,
-			localize('conflicts detected', "Conflicts Detected"),
-			[
-				localize('show conflicts', "Show Conflicts"),
-				localize('replace local', "Replace Local"),
-				localize('replace remote', "Replace Remote"),
-				localize('cancel', "Cancel"),
+		await this.dialogService.prompt({
```

Comment:
@sandy081 fyi please check, this one is a bit more complex adoption of the new dialog api